File: src/dataset/javascript/isogram_exercism-ThomasZumsteg/claude/ClaudeSonnet4_isogram_v3.js
class Isogram {
    constructor(word) {
        this.word = word.toLowerCase();
    }

    isIsogram() {
        const seen = new Set();
        for (let i = 0; i < this.word.length; i++) {
            const char = this.word[i];
            if (char !== '-' && char !== ' ') {
                if (seen.has(char)) {
                    return false;
                }
                seen.add(char);
            }
        }
        return true;
    }
}

module.exports = Isogram;

File: src/dataset/cpp/isogram_exercism-Akshive/claude/ClaudeSonnet4_isogram_v3.cpp
#include "isogram.h"
#include <string>
#include <bitset>

namespace isogram {

    bool is_isogram(std::string word) {
        std::bitset<26> seen;
        
        for (char c : word) {
            if (c >= 'A' && c <= 'Z') {
                c += 32;
            } else if (c < 'a' || c > 'z') {
                continue;
            }
            
            int index = c - 'a';
            if (seen[index]) {
                return false;
            }
            seen[index] = 1;
        }
        
        return true;
    }

}